app/react/kubernetes/ServicesView/ServicesDatatable/columns/externalIP.tsx

```diff
@@ -0,0 +1,109 @@
+import { CellProps, Column } from 'react-table';
+
+import { Service } from '../../types';
+
+import { ExternalIPLink } from './externalIPLink';
+
+// calculate the scheme based on the ports of the service
+// favour https over http.
+function getSchemeAndPort(svc: Service): [string, number] {
+  let scheme = '';
+  let servicePort = 0;
+
+  svc.Ports?.forEach((port) => {
+    if (port.Protocol === 'TCP') {
+      switch (port.TargetPort) {
+        case '443':
+        case '8443':
+        case 'https':
+          scheme = 'https';
+          servicePort = port.Port;
+          break;
+
+        case '80':
+        case '8080':
+        case 'http':
+          if (scheme !== 'https') {
+            scheme = 'http';
+            servicePort = port.Port;
+          }
+          break;
+
+        default:
+          break;
+      }
+    }
+  });
+
+  return [scheme, servicePort];
+}
+
+export const externalIP: Column<Service> = {
+  Header: 'External IP',
+  id: 'externalIP',
+  accessor: (row) => row.IngressStatus?.slice(0),
+  Cell: ({ row }: CellProps<Service>) => {
+    const isExternalName = row.original.Type === 'ExternalName';
+    if (isExternalName) {
+      if (!row.original.ExternalName) {
+        return '-';
+      }
+
+      const linkto = `http://${row.original.ExternalName}`;
```

Comment:
any way to see if we should use https?

Reply:
No not really a nice way to do it.  You don't know the port.  
AFAIK it's purely like a dns entry that can be used by pods in the cluster.